pkg/opensearch/response_parser.go

```diff
@@ -1181,18 +1181,16 @@ func (rp *responseParser) trimDatapoints(frames *data.Frames, target *Query) {
 		return
 	}
 
-	trimEdges, err := histogram.Settings.Get("trimEdges").Int()
+	trimEdges, err := castToInt(histogram.Settings.Get("trimEdges"))
 	if err != nil {
 		return
 	}
 
 	for _, f := range *frames {
 		if f.Rows() > trimEdges*2 {
 			for i := 0; i < trimEdges; i++ {
-				f.DeleteRow(i)
-			}
-			for i := f.Rows() - trimEdges; i < f.Rows(); i++ {
-				f.DeleteRow(i)
```

Comment:
With the previous logic, this series of data points `| 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 |` and `Trim Edges` set to 2 would be trimmed to `| 1 | 3 | 5 | 6 | 7 | 9 |`.

With the fixed logic, it would be trimmed to `| 3 | 4 | 5 | 6 | 7 |`.

The updated logic just trims from the beginning and end of the series instead of incrementing an index which ends up skipping data points that should be trimmed and not trimming enough data points from the end of the series.